target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -123,3 +123,13 @@ platform_do_upgrade() {
 		;;
 	esac
 }
+
+platform_pre_upgrade() {
```

Comment:
My understanding is that routine is going away. See a074ed293, e98eab1b6, 3b30ff29 and so on.

In any case this belongs to the subtarget, see target/linux/ath79/mikrotik/base-files/lib/upgrade/platform.sh